src/nginx/t/api_basepath.t

```diff
@@ -0,0 +1,189 @@
+
+my $t = Test::Nginx->new()->has(qw/http proxy/)->plan(18);
+
+# Save service name in the service configuration protocol buffer file.
+
+$t->write_file('service.pb.txt', ApiManager::get_bookstore_service_config . <<"EOF");
+control {
+  environment: "http://127.0.0.1:${ServiceControlPort}"
+}
+EOF
+
+$t->write_file('server.pb.txt', <<"EOF");
+api_service_config {
+  base_path: "/api"
+}
+EOF
+
+
+ApiManager::write_file_expand($t, 'nginx.conf', <<"EOF");
+%%TEST_GLOBALS%%
+daemon off;
+events {
+  worker_connections 32;
+}
+http {
+  %%TEST_GLOBALS_HTTP%%
+  server_tokens off;
+  server {
+    listen 127.0.0.1:${NginxPort};
+    server_name localhost;
+    location / {
+      endpoints {
+        api service.pb.txt;
+        %%TEST_CONFIG%%
+        server_config server.pb.txt;
+        on;
+      }
+      proxy_pass http://127.0.0.1:${BackendPort};
+    }
+  }
+}
+EOF
+
+$t->run_daemon(\&bookstore, $t, $BackendPort, 'bookstore.log');
+$t->run_daemon(\&servicecontrol, $t, $ServiceControlPort, 'servicecontrol.log');
+is($t->waitforsocket("127.0.0.1:${BackendPort}"), 1, 'Bookstore socket ready.');
+is($t->waitforsocket("127.0.0.1:${ServiceControlPort}"), 1, 'Service control socket ready.');
+$t->run();
+
+################################################################################
+
+my $response_first = ApiManager::http_get($NginxPort,'/api/shelves?key=this-is-an-api-key');
+my $response_second = ApiManager::http_get($NginxPort,'/shelves?key=this-is-an-api-key');
+
+$t->stop_daemons();
+
+my ($response_headers, $response_body) = split /\r\n\r\n/, $response_first, 2;
+
+like($response_headers, qr/HTTP\/1\.1 200 OK/, 'Returned HTTP 200.');
+is($response_body, <<'EOF', 'Shelves returned in the response body.');
+{ "shelves": [
+    { "name": "shelves/1", "theme": "Fiction" },
+    { "name": "shelves/2", "theme": "Fantasy" }
+  ]
+}
+EOF
+
+($response_headers, $response_body) = split /\r\n\r\n/, $response_second, 2;
+
+like($response_headers, qr/HTTP\/1\.1 200 OK/, 'Returned HTTP 200.');
+is($response_body, <<'EOF', 'Shelves returned in the response body.');
+{ "shelves": [
+    { "name": "shelves/1", "theme": "Fiction" },
+    { "name": "shelves/2", "theme": "Fantasy" }
+  ]
+}
+EOF
+
+my @requests = ApiManager::read_http_stream($t, 'bookstore.log');
+is(scalar @requests, 2, 'Backend received two requests');
+
+my $r = shift @requests;
+is($r->{verb}, 'GET', 'Backend request was a get');
+is($r->{uri}, '/shelves?key=this-is-an-api-key', 'Backend uri was /shelves');
+is($r->{headers}->{host}, "127.0.0.1:${BackendPort}", 'Host header was set');
+
+$r = shift @requests;
+is($r->{verb}, 'GET', 'Backend request was a get');
+is($r->{uri}, '/shelves?key=this-is-an-api-key', 'Backend uri was /shelves');
+is($r->{headers}->{host}, "127.0.0.1:${BackendPort}", 'Host header was set');
+
+
+
+@requests = ApiManager::read_http_stream($t, 'servicecontrol.log');
+is(scalar @requests, 1, 'Service control received one request');
```

Comment:
why only 1 call?   second Check is cached?
how about report?  Report may come in some race condition

Please disable check and report cache,  in order not to have flaky test.

Reply:
Disabled check cache and added the report_done file waiting code to avoid flaky test